Lib/test/_test_multiprocessing.py

```diff
@@ -2611,6 +2620,41 @@ def errback(exc):
         p.close()
         p.join()
 
+    def test_broken_process_pool1(self):
+        from multiprocessing.pool import BrokenProcessPool
+        p = multiprocessing.Pool(2)
+        res = p.map_async(waiting, range(10))
+        # Kill one of the pool workers.
+        waiting(None)
+        pid = p._pool[0].pid
+        os.kill(pid, signal.SIGTERM)
+        with self.assertRaises(BrokenProcessPool):
+            res.get()
```

Comment:
What happens if I try to start a new task after the pool is broken? Probably worth checking.

Reply:
Added one test for this.